src/hooks/use-deck-editor.ts

```diff
@@ -68,14 +68,18 @@ export const useDeckEditor = (deck: Deck): DeckEditorReturn => {
       const updatedCards = Object.values(state.updatedCards).filter(filterBlankCards);
       const deletedCards = Object.values(state.deletedCards);
 
-      const promises = [
-        decksClient.updateDeckById(state.currentDeck),
+      // alone to prevent race condition on backend since updating cards mutates a deck as well
```

Comment:
actually gonna revert this and fix it on the backend side bc i've identified the cause now